ocaml/idl/datamodel_cluster.ml

```diff
@@ -194,6 +194,16 @@ let t =
              (Some (VString Constants.default_smapiv3_cluster_stack))
            "Simply the string 'corosync'. No other cluster stacks are \
             currently supported"
+       ; field ~qualifier:DynamicRO ~lifecycle:[] ~ty:Bool "is_quorate"
+           ~default_value:(Some (VBool false)) "Whether the cluster is quorate"
+       ; field ~qualifier:DynamicRO ~lifecycle:[] ~ty:(Set String)
```

Comment:
This is a `string set`, which ends up getting populated with IP addresses if I see it right. In the context of the XenAPI, it may be better to have `host` references or `cluster_host` references. Still better may be to record this liveness state in a field in the relevant `cluster_host`.

Reply:
There seems to be a `joined` field in the datamodel already, maybe we can use this one?
```ocaml
    ; field ~qualifier:StaticRO ~lifecycle ~ty:Bool "joined"
           ~default_value:(Some (VBool true))
           "Whether the cluster host has joined the cluster"
```

Reply:
Yeah I saw that and am not clear on what it means exactly. Immediately below there is this comment:
```
         (* TODO: add `live` member to represent whether corosync believes that this
                  cluster host actually is enabled *)
```

Reply:
Seems like we are doing the TODO here, but I feel we can reuse the `joined` field though.

Reply:
Perhaps yes, but we need to check how exactly it is used at the moment.

Reply:
I have decided to introduce a new field called `live` as well as a corresponding `last_update_live` field to make sure that the user is not confused by stale info. I feel using `joined` field complicates the current situation with other methods that can modify join field such as `destroy` and `forget`. The `live` field itself cannot be modified by any API functions, which makes it a bit simpler.